### PR TITLE
Make change of options.yml backward compatible

### PR DESCRIPTION
**Pull request by bgeuken**
This adds support for the old format of options.yml files.

Follow up of #5622

### CODE REVIEW

**Comment by vpereira** [comment]
@bgeuken do we have to change something in the ```rake docker:build``` task or in the ```docker-compose run --no-deps --rm frontend bundle exec rake dev:bootstrap RAILS_ENV=development``` ?

for me even after generate the new file with ```(cd src/api/; rake migrate_options_yml)``` , I get the following trace:

```
rake aborted!
NoMethodError: undefined method `[]' for nil:NilClass
/obs/src/api/config/environment.rb:17:in `<top (required)>'
Tasks: TOP => dev:bootstrap => environment
(See full trace by running task with --trace)
docker-compose stop
Stopping openbuildservice_db_1 ... done
rake aborted!
Command failed with status (1): [docker-compose run --no-deps --rm frontend...]
/home/vpereira/open-build-service/Rakefile:34:in `block (2 levels) in <top (required)>'
/home/vpereira/.rvm/gems/ruby-2.5.0/gems/rake-12.3.1/exe/rake:27:in `<top (required)>'
/home/vpereira/.rvm/gems/ruby-2.5.0/bin/ruby_executable_hooks:15:in `eval'
/home/vpereira/.rvm/gems/ruby-2.5.0/bin/ruby_executable_hooks:15:in `<main>'
Tasks: TOP => docker:build
(See full trace by running task with --trace)
```

**Comment by bgeuken** [comment]
> vpereira commented 2 hours ago 
> for me even after the migration with (cd src/api/; rake migrate_options_yml),

Could it be that you switched back to a branch that does not have the recent changes from `master`? In that case this PR should solve this.

File: src/api/config/environment.rb [hunk 1]
```diff
@@ -6,7 +6,14 @@
 path = Rails.root.join('config', 'options.yml')
 
 begin
-  CONFIG = YAML.load_file(path)[Rails.env]
+  config = YAML.load_file(path)
+  if config.key?(Rails.env)
+    CONFIG = config[Rails.env]
+  else
+    # FIXME: Remove with the next stable release (v2.10 or v3.0)
+    puts "DEPRECATED: Please update your options.yml by running 'rake migrate_options_yml'"
```
Comment (ChrisBr) [inline src/api/config/environment.rb]: I would prefer using Rails.logger!

**Comment by bgeuken** [comment]
@ChrisBr updated

**Comment by hennevogel** [comment]
@bgeuken this happens if you rebuild the image and you do not have migrated the `options.yml`. Which you need the image for: :chicken: :egg: :fire:

**Comment by hennevogel** [comment]
For the development environment it's enough to copy over the example file again...

**Comment by bgeuken** [comment]
@hennevogel Okay, but this should be solved with this PR. The error was actually caused by #5622

So, can we merge this?:smile:

File: src/api/config/environment.rb [hunk 2]
```diff
@@ -6,7 +6,14 @@
 path = Rails.root.join('config', 'options.yml')
 
 begin
-  CONFIG = YAML.load_file(path)[Rails.env]
+  config = YAML.load_file(path)
+  if config.key?(Rails.env)
+    CONFIG = config[Rails.env]
+  else
+    # FIXME: Remove with the next stable release (v2.10 or v3.0)
+    Rails.logger.info "DEPRECATED: Please update your options.yml by running 'rake migrate_options_yml'"
```
Comment (ChrisBr) [inline src/api/config/environment.rb]: This doesnt work as you don't have a logger in the initializer! You can either define it with:

```
logfile = File.open("#{Rails.root}/log/#{Rails.env}.log", 'a')  
logger = Logger.new(logfile)
```

or leave the puts!

**Comment by ChrisBr** [comment]
Argh! This does not work :cry:

**Comment by hennevogel** [comment]
What does not work?

**Comment by bgeuken** [comment]
Using Rails.logger -> #5653

**Pull request merged**